### PR TITLE
Make use of rcutils log severity defined enum instead of duplicating code

### PR DESCRIPTION
**Pull request by BMarchi**
Follow up of https://github.com/ros2/rclpy/pull/458. Remove log severity hard coded and copied values from rcutils

### CODE REVIEW

**Comment by BMarchi** [comment]
- Linux [Build Status](https://ci.ros2.org/job/ci_linux/8740/)
- Arch [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/4569/)
- macOS [Build Status](http://ci.ros2.org/job/ci_osx/7142/)
- Windows [Build Status](http://ci.ros2.org/job/ci_windows/8654/)

File: rclpy/src/rclpy/_rclpy_logging.c [hunk 1]
```diff
@@ -199,6 +199,54 @@ rclpy_logging_severity_level_from_string(PyObject * Py_UNUSED(self), PyObject *
   return PyLong_FromLongLong(severity);
 }
 
+/// Get log unset severity level as int from rcutils
+/// \return RCUTILS_LOG_SEVERITY_UNSET
```
Comment (hidmic) [inline rclpy/src/rclpy/_rclpy_logging.c]: @BMarchi nit^2: `\return RCUTILS_LOG_SEVERITY_UNSET as a PyLong`, here and everywhere else.

**Comment by dirk-thomas** [comment]
The patch is fine for now.

I am still worried that this will get us further away from being able to import `rclpy` modules side effect free (see #420). But that shouldn't hold up this PR.

**Comment by hidmic** [comment]
> I am still worried that this will get us further away from being able to import `rclpy` modules side effect free (see #420). But that shouldn't hold up this PR.

Yes, I remember the ticket and the discussions. Thanks for bumping that this one up. I also recall that most variations trying to circumvent the problem weren't great either. It'd be great if we could spend the time to really solve the problem i.e. C extensions need not be special, it's a trap to make them so.

**Pull request merged**